src/main/scala/konstructs/main.scala

```diff
@@ -1,13 +1,71 @@
 package konstructs
 
+import java.util.UUID
+import java.io.{ FileReader, FileNotFoundException, File }
+import org.apache.commons.io.FileUtils
+import scala.collection.JavaConverters._
 import akka.actor.ActorSystem
 import com.typesafe.config.ConfigFactory
+import com.google.gson.reflect.TypeToken
+import konstructs.api.{ GsonDefault, Position }
 import konstructs.plugin.PluginLoaderActor
 
 object Main extends App {
   val conf =
     ConfigFactory.parseFile(new java.io.File("konstructs.conf")).withFallback(ConfigFactory.load())
   implicit val system = ActorSystem("main", conf)
+  // Run any global backwards compatibility scripts
+  // Plugins should handle their own backwards compatibility
+  Compatibility.process()
   val loader = system.actorOf(PluginLoaderActor.props(conf.getConfig("konstructs")), "plugin-loader")
   loader ! PluginLoaderActor.Start
 }
+
+object Compatibility {
+  import ShardActor.{ shardId, positionMappingFile }
+  val TypeOfPositionMapping = new TypeToken[java.util.Map[String, UUID]](){}.getType
+
+  def process() {
+    println("STARTING MIGRATION PROCESS, DON'T INTERRUPT!")
+    handleOldUuidDB()
+    println("MIGRATION PROCESS FINISHED")
+  }
+
+  val PositionRegex = """(-?\d*)-(-?\d*)-(-?\d*)""".r
+
+  def parsePositionStr(posStr: String): Position = posStr match {
+    case PositionRegex(x, y, z) => new Position(x.toInt, y.toInt, z.toInt)
+  }
+
+  /*
+   * This "migration" only works if disk storage on default location was used
+   */
+  def handleOldUuidDB() {
+    val gson = GsonDefault.getDefaultGson
+    try {
+      val fileName = s"meta/org/konstructs/block-manager/position-mapping.json"
+      val reader = new FileReader(fileName)
+      val mapping: java.util.Map[String, UUID] = gson.fromJson(reader, TypeOfPositionMapping)
```

Comment:
Then you end up in the catch clause, skipping the process. Either it's a new server or it was already migrated.